src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1187,21 +1179,60 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     const common = this.fallback
       ? this.fallback.getCommonForBlockNumber(
           this.common,
-          BigInt(newBlock.header.number.toString())
+          BigInt(block.header.number.toString())
         )
       : this.common;
 
     const vm = await VM.create({
-      state: trie,
       activatePrecompiles: false,
       common,
       allowUnlimitedContractSize: this.vm.allowUnlimitedContractSize,
       blockchain,
       stateManager: this.fallback
         ? new ForkStateManager({ common, trie: trie as ForkTrie })
```

Comment:
With `debug_storageRangeAt` [throwing](https://github.com/trufflesuite/ganache/pull/3470/files#diff-33bf12664f527a9e8ee226d4340fc6c3255c690db880c187f0a69166dd24e55fR2995) for the mentioned reason, would `ForkStateManager` be reachable? 

`traceTransaction` is the only other place to use `fastForwardVM` but it has an existing TODO for mainnet forking. I'm guessing this is benign for now but something we eventually want on mainnet forking.